# Explorar datos FIFA casos especial
## Fecha: 05/Septiembre/2022
## Autor:
    LUIS ALBERTO JIMENEZ SOTO y EZEQUIEL OLIVAS MELENDEZ
# Objetivo
Explorar datos de jugadores de la FIFA para contestar una serie de preguntas
# Desarrollo
## Cargar librerias

In [18]:
import pandas as pd
import matplotlib.pyplot as plt

## Mostrar las primeras 10 observaciones

In [19]:
# Cargamos el dataset
datos = pd.read_csv("https://raw.githubusercontent.com/rpizarrog/Analisis-Inteligente-de-datos/main/datos/datos.FIFA.limpios.csv", encoding='latin1')
datos.head(10)

Unnamed: 0               Name  Age Nationality  Overall  Potential  \
0           1           L. Messi   31   Argentina       94         94   
1           2  Cristiano Ronaldo   33    Portugal       94         94   
2           3          Neymar Jr   26      Brazil       92         93   
3           4             De Gea   27       Spain       91         93   
4           5       K. De Bruyne   27     Belgium       91         92   
5           6          E. Hazard   27     Belgium       91         91   
6           7          L. Modric   32     Croatia       91         91   
7           8          L. Suárez   31     Uruguay       91         91   
8           9       Sergio Ramos   32       Spain       91         91   
9          10           J. Oblak   25    Slovenia       90         93   

                  Club Preferred.Foot  International.Reputation  Weak.Foot  \
0         FC Barcelona           Left                       5.0        4.0   
1             Juventus          Right                       5.0        4.0   
2  Paris Saint-Germain          Right                       5.0        5.0   
3    Manchester United          Right                       4.0        3.0   
4      Manchester City          Right                       4.0        5.0   
5              Chelsea          Right                       4.0        4.0   
6          Real Madrid          Right                       4.0        4.0   
7         FC Barcelona          Right                       5.0        4.0   
8          Real Madrid          Right                       4.0        3.0   
9      Atlético Madrid          Right                       3.0        3.0   

   ...  StandingTackle SlidingTackle GKDiving  GKHandling  GKKicking  \
0  ...            28.0          26.0      6.0        11.0       15.0   
1  ...            31.0          23.0      7.0        11.0       15.0   
2  ...            24.0          33.0      9.0         9.0       15.0   
3  ...            21.0          13.0     90.0        85.0       87.0   
4  ...            58.0          51.0     15.0        13.0        5.0   
5  ...            27.0          22.0     11.0        12.0        6.0   
6  ...            76.0          73.0     13.0         9.0        7.0   
7  ...            45.0          38.0     27.0        25.0       31.0   
8  ...            92.0          91.0     11.0         8.0        9.0   
9  ...            12.0          18.0     86.0        92.0       78.0   

   GKPositioning  GKReflexes      Valor  Estatura  PesoKgs  
0           14.0         8.0  110500000      1.70    72.12  
1           14.0        11.0   77000000      1.88    83.01  
2           15.0        11.0  118500000      1.75    68.04  
3           88.0        94.0   72000000      1.93    76.20  
4           10.0        13.0  102000000      1.80    69.85  
5            8.0         8.0   93000000      1.73    73.94  
6           14.0         9.0   67000000      1.73    66.22  
7           33.0        37.0   80000000      1.83    86.18  
8            7.0        11.0   51000000      1.83    82.10  
9           88.0        89.0   68000000      1.88    87.09  

[10 rows x 50 columns]

## Describir estructura de los datos

In [20]:
datos.to_json

<bound method NDFrame.to_json of        Unnamed: 0                Name  Age Nationality  Overall  Potential  \
0               1            L. Messi   31   Argentina       94         94   
1               2   Cristiano Ronaldo   33    Portugal       94         94   
2               3           Neymar Jr   26      Brazil       92         93   
3               4              De Gea   27       Spain       91         93   
4               5        K. De Bruyne   27     Belgium       91         92   
...           ...                 ...  ...         ...      ...        ...   
17950       17951        J. Lundstram   19     England       47         65   
17951       17952  N. Christoffersson   19      Sweden       47         63   
17952       17953           B. Worman   16     England       47         67   
17953       17954      D. Walker-Rice   17     England       47         66   
17954       17955           G. Nugent   16     England       46         66   

                      Club Pre

## Resolver e interpretar cada una de las 10 preguntas

### ¿Qué jugadores ganan más dinero, el top 5?

In [21]:
valor = datos.sort_values(['Valor'], ascending=False)
valor[["Name", "Valor"]].head()

Name      Valor
2      Neymar Jr  118500000
0       L. Messi  110500000
4   K. De Bruyne  102000000
5      E. Hazard   93000000
15     P. Dybala   89000000

### ¿Cuántos jugadores de fútbol son izquierdos?

In [22]:
frecuencia_pie = (datos.groupby("Preferred.Foot").agg(frequency=("Preferred.Foot", "count")))
type(frecuencia_pie)
# frecuencia_nac
frecuencia_pie = frecuencia_pie.sort_values(['frequency'], ascending=True)
frecuencia_pie

frequency
Preferred.Foot           
Left                 4159
Right               13748

### ¿Cuál es el valor medio (promedio) de estaura en cms. o metros de jugadores de fútbol?

In [23]:
estatura_promedio = round(datos["Estatura"].mean(),2)

print("La estatura promedio de los jugadores es de",estatura_promedio,"m")

La estatura promedio de los jugadores es de 1.81 m


### ¿Cómo se llaman los jugadores más bajitos en estatura?

In [24]:
estaturas_sorted = datos.sort_values(['Estatura'], ascending=True)
estaturas_sorted.head()[["Name", "Estatura"]]

Name  Estatura
6474    H. Nakagawa      1.55
6172     N. Barrios      1.55
16914  K. Yamaguchi      1.55
2732          Élton      1.57
2462       J. Plata      1.57

### ¿De qué nacionalidad o país hay más jugadores registrados?

In [25]:
frec_nacionalidad = (datos.groupby("Nationality")).agg(frequency=("Nationality", "count"))
type(frec_nacionalidad)
frec_nacionalidad = frec_nacionalidad.sort_values(['frequency'], ascending=False)
frec_nacionalidad.head(1)

frequency
Nationality           
England           1660

# Interpretación

1. ¿Qué jugadores ganan más dinero (Top 5)?

        - Neymar Jr
        - L. Messi
        - K. De Bruyne
        - E. Hazard
        - P. Dybala

2. ¿Cuantos jugadores de futbol son izquierdos?

        4159


3. ¿Cual es el valor medio (promedio) de estatura en cms o metros de jugadores de futbol?

        La estatura promedio de los jugadores es de 1.81 m

4. ¿Como se llaman los jugadores mas bajitos en estatura (top 5)?

        - H. Nakagawa

        - N. Barrios

        - K. Yamaguchi

        - Élton

        - J. Plata

5. ¿De que pais hay mas jugadores registrados y cuantos?

        England 1660
